python/tvm/driver/tvmc/compiler.py

```diff
@@ -196,9 +196,9 @@ def compile_model(
     for codegen_from_cli in extra_targets:
         codegen = composite_target.get_codegen_by_target(codegen_from_cli["name"])
         partition_function = codegen["pass_pipeline"]
-        mod = partition_function(mod, params)
+        mod, codegen_config = partition_function(mod, params)
```

Comment:
In the interest of standardizing all the `partition_for_*` functions, I think it would be good to do a little refactoring in the `partition_for_tensorrt` so that it only returns `mod`. This is important not only for TVMC, but I think it would be a good move in terms of future partitioning support in TVM.

There is some new functionality (yet to be merged) in #7577 to have an `init` function per target in tvmc, in which we can hen populate the configs using the existing `get_tensorrt_version`.

I suggest we coordinate to get #7577 landed and then adjust this to comply with the suggestions above, so that we can get a uniform support for all targets. What do you think?

(cc @comaniac)

Reply:
I agree that it would be easier to ask all pritition functions return a single module. However, the init function introduced by #7577 seems not applicable to this case to me. The config required by TensorRT is used when building the module; while the init function used by Vitis-AI imports the required package. I have no idea how could we make it general enough to support both Vitis-AI and TensorRT.

On the other hand, let a partition function return a tuple of module and build config seems more straightforward at the first glance, as I cannot think of a better idea for now. First, partition functions won't be a Relay pass in a pass sequence (at least for now). Second, it is reasonable that the build config may need to be constructed during the partition. Separating them to two APIs may introduce overheads and be more confusing to non TVMC users, which are still majority in this community.

Reply:
I agree with @comaniac, I don't see a good way to separate this yet. The parameters used for `parition_for_tensorrt` affect both partitioning and building the module.

Reply:
Fair enough. I thought a bit more about it and as with #7577, we will be already sending a config dictionary for the partition function. so I think it is ok to get the same, or a modified version of that dictionary as a result, which fits with the existing support on TensorRT.

Reply:
I guess an alternative could be for the partitioning to be done within the pass context, and have the partition function set the config in the passcontext.

Reply:
That's our ultimate goal of the composite target, which integrates partition and other codegen specific functions. Once we have it, the partition logic could be completely hidden from not only users but also frontend interface.

Reply:
For now, I tried to receive and apply the config returned partiton_for_tensrort.

Reply:
Yeah, I agree this is simpler and effective from now. Happy to approve once we get the linting issues sorted out and CI passing here.